### DISCUSSION TITLE
Github copilot is good!

### DISCUSSION THREAD

**Discussion by ghost** [community/community#14087] [Copilot Conversations]
This project will be great once it comes out of technical preview. I know it. It's so helpful. One time I didnt know how to read a file in javascript, and when I wrote a function named `function readFile(dir, name, ext) {}` it really helped me.

Took a while to begin showing inline suggestions. But once it began working it was really good.

Nice autocomplete I like it

**Comment by MihailCosmin** [comment]
I totally agree. I am completely satisfied with what I have seen so far.

I am using it mostly for python and it was amazing so far.
I wanted to increase letters from A to Z inside a for loop. And I knew the logic, but didn't knew how to do it in python. I thought I would have to search on stack overflow. But the copilot correctly filled the piece of code in a second.

It sometimes even appears to read my mind when trying to write some descriptive comments.

For me it will definitely improve my coding experience, since I am not a total expert.

**Comment by Urethramancer** [comment]
It's been a great tool for API discovery. When using an unfamiliar SDK, it can help me get started. Or even suggest a different one if I just write what I want in a comment. Of course, sometimes it just stares blankly, but it's not magical. Yet.

When it fills in whole multi-line functions, it's good at remembering those closing statements people may not know about (some DB packages require closing row lists in Go, for example, and only one end of HTTP ops requires it).

And yeah, it really feels like it's reading my mind. I sit down to continue on a project, find the file where I want to add something, press enter to start on a new functiom, and sometimes it suggests exactly what I need before I've even typed the comment marker! Of course, once you have "get" functions, it should be obvious you may want "set", "modify" and "delete" too, but it still seems magical at times.

Occasionally it'll make some scary accurate guesses about what I want to add to a blank function where the function name doesn't necessarily give any context. It seems to have learned some of my internal packages pretty well already too.
